src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionResultSimulationService.java

```diff
@@ -0,0 +1,127 @@
+        programmingExerciseStudentParticipation.setInitializationDate(ZonedDateTime.now());
+        programmingExerciseStudentParticipation.setProgrammingExercise(programmingExercise);
+        participationRepository.save(programmingExerciseStudentParticipation);
+        return programmingExerciseStudentParticipation;
+    }
+
+    /**
+     * This method creates a new submission for the provided user
+     * @param exerciseId the exerciseId of the exercise for which a submission should be created
+     * @return the newly created and stored submission
+     */
+    public ProgrammingSubmission createSubmission(Long exerciseId) {
+        User user = userService.getUserWithGroupsAndAuthorities();
+        Participant participant = user;
+        ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation;
+        ProgrammingExercise programmingExercise = programmingExerciseService.findByIdWithEagerStudentParticipationsAndSubmissions(exerciseId);
+        Optional<StudentParticipation> optionalStudentParticipation = participationService.findOneByExerciseAndStudentLoginWithEagerSubmissionsAnyState(programmingExercise,
+                user.getLogin());
+        if (optionalStudentParticipation.isEmpty()) {
+            programmingExerciseStudentParticipation = createParticipation(programmingExercise, participant, user);
+        }
+        else {
+            programmingExerciseStudentParticipation = (ProgrammingExerciseStudentParticipation) optionalStudentParticipation.get();
+        }
+
+        ProgrammingSubmission programmingSubmission = new ProgrammingSubmission();
+        programmingSubmission.setCommitHash(VCSSimulationUtils.simulateCommitHash());
+        programmingSubmission.setSubmitted(true);
+        programmingSubmission.setSubmissionDate(ZonedDateTime.now());
+        programmingSubmission.setType(SubmissionType.MANUAL);
+        programmingExerciseStudentParticipation.addSubmissions(programmingSubmission);
+
+        programmingSubmissionRepository.save(programmingSubmission);
+        return programmingSubmission;
+    }
+
+    /**
+     *  This method creates a new result for the provided participation
+     * @param programmingExerciseStudentParticipation the participation for which the new result should be created
+     * @return the newly created and stored result
+     */
+    public Result createResult(ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation) {
+        Optional<ProgrammingSubmission> programmingSubmission = programmingSubmissionRepository
+                .findFirstByParticipationIdOrderBySubmissionDateDesc(programmingExerciseStudentParticipation.getId());
+        Result result = new Result();
+        result.setSubmission(programmingSubmission.get());
+        result.setParticipation(programmingExerciseStudentParticipation);
+        result.setRated(true);
+        result.resultString("7 of 13 passed");
+        result.score(54L);
```

Comment:
6 feedback objects are missing with the corresponding test case name and the error message

Reply:
i added the feedback objects